src/axom/sidre/core/View.hpp

```diff
@@ -208,6 +209,23 @@ class View
    */
   bool isString() const { return m_state == STRING; }
 
+  /*!
+   * \brief Return whether view data is on device, as determined
+   * by Axom's memory management.
+   *
+   * By convention, this returns a false if data is not allocated.
+   */
+  bool isDeviceData() const;
+
+  /*!
+   * \brief Return whether view data is accessible on the host CPU,
+   * as determined by Axom's memory management.
+   *
+   * By convention, this returns a false if data is not allocated,
+   * because we expect null pointers to be correctly checked before use.
+   */
+  bool isHostAccessible() const;
```

Comment:
what about external views?

Reply:
Same as checking for device data.  Axom looks at the pointer to determine it's memory space and associated characteristics.